docs/modules/ROOT/pages/references/parameters.adoc

```diff
@@ -6,14 +6,207 @@ The parent key for all of the following parameters is `vault`.
 
 [horizontal]
 type:: string
-default:: `syn-vault`
+default:: `vault`
 
 The namespace in which to deploy this component.
 
 
-== Example
+== `name`
 
+[horizontal]
+type:: string
+default:: `vault`
+
+The name of the deployed component.
+
+== `images`
+
+[horizontal]
+type:: dictionary
+
+Dictionary containing the container images used by this component.
+
+== `charts`
+
+[horizontal]
+type:: dictionary
+
+Dictionary containing the chart versions used by this component.
+
+== `ingress.enabled`
+
+[horizontal]
+type:: bool
+default:: `true`
+
+Whether to create an ingess object.
+
+== `ingress.host`
+
+[horizontal]
+type:: string
+default:: `vault.todo.tld`
+
+Defines the FQDN of the ingress, should be overwritten on the cluster level.
+
+== `ingress.annotations`
+
+[horizontal]
+type:: dict
+default:: {}
+
+The annotations added to the created ingress
+Needs to be set according to the deployed ingress controller.
+
+
+Example:
 [source,yaml]
 ----
-namespace: example-namespace
+annotations:
+  cert-manager.io/cluster-issuer: letsencrypt-production
+  kubernetes.io/ingress.class: nginx
+----
+
+== `storage.size`
+
+[horizontal]
+type:: string
+default:: `10G`
+
+The requested storage size for secret storage.
+
+== `storage.class`
+
+[horizontal]
+type:: string
+default:: ``
+
+The kubernetes storage class to request.
+
+== `resources`
+
+[horizontal]
+type:: dict
+default::
++
+[source,yaml]
 ----
+requests:
+  memory: 256Mi
+  cpu: 250m
+limits:
+  memory: 512Mi
+  cpu: 1000m
+----
+
+The resource requests and limits.
+
+== `config`
+
+[horizontal]
+type:: dict
+default::
++
+[source,yaml]
+----
+policies:
+  - name: backup
+    rules: |
+      path "sys/storage/raft/snapshot" {
+        capabilities = ["read"]
+      }
+secrets:
+  - type: kv
+    path: clusters/kv
+    description: General secrets for clusters
+    options:
+      version: 2
+auth:
+  - type: kubernetes
+    roles:
+      - name: backup
+        bound_service_account_names: '${vault:name}-backup'
+        bound_service_account_namespaces: ${vault:namespace}
+        policies: backup
+        ttl: 1h
+----
+
+The configuration for vault.
+The default configuration adds a general key-value secret store and a default backup user.
+If this backup user isn't present, backups using k8up won't succeed.
+This configuration may directly contain secrets.
```

Comment:
You probably want to clarify that the config may directly contain secret references rather than secrets